File: src/main/java/com/example/HealthAndFitnessPlatform/model/Like.java
package com.example.HealthAndFitnessPlatform.model;


import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "likeTBL")
@Data
@AllArgsConstructor
@RequiredArgsConstructor
public class Like {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "userId")
    private User user;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "recipeId")
    private Recipe recipe;

    @Column(name = "createdAt")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime createdAt = LocalDateTime.now();


    @PrePersist
    public void prePersist() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
    }
}

File: HealthAndFitnessPLatformNG/src/app/model/like.ts
export class Like {

     likeId ?: number;
     userId : number;
     recipeId : number;
     createdAt ?: string;

     constructor(likeId:number,userId:number,recipeId:number,createdAt:string){
               this.likeId = likeId;
               this.userId =userId;
               this.recipeId = recipeId;
               this.createdAt = createdAt;
     }

}